doc/how_to/migrate/from_streamlit.md

```diff
@@ -0,0 +1,988 @@
+
+#### Panel Session Cache Example
+
+```python
+from time import sleep
+
+import numpy as np
+import panel as pn
+from matplotlib.figure import Figure
+
+
+def get_data():
+    print("get_data func")
+    sleep(1.0)
+    return np.random.normal(1, 1, size=100)
+
+def plot(data, bins):
+    print("plot func", bins)
+    sleep(2)
+    fig = Figure(figsize=(8,4))
+    ax = fig.subplots()
+    ax.hist(data, bins=bins)
+    return fig
+
+pn.extension(sizing_mode="stretch_width", template="bootstrap")
+
+data = get_data()
+bins = pn.widgets.IntSlider(value=20, start=10, end=30, step=1)
+cplot = pn.cache(plot)
+bplot = pn.bind(cplot, data, bins)
+pn.Column(bins, pn.panel(bplot, loading_indicator=True)).servable()
+```
+
+Please note that using `pn.cache` on `get_data` would not make a
+difference since `get_data` is only run once per user session.
+
+The app looks like
+
+![Panel Session Cache Example](https://user-images.githubusercontent.com/42288570/245366569-d1808e7b-812f-4044-8bad-73205027fd63.gif)
+
+You can also use `pn.cache` as an annotation similar to `st.cache_data`. I.e. as
+
+```python
+@pn.cache
+def plot(data, bins):
+    ...
+```
+
+But be aware the the annotation approach has a tendency to mix up your business logic
+(`data` and `plot` function) and your caching logic (when and how to apply caching). This can make
+your code harder to maintain and reuse for other use cases.
+
+### Global Cache Example
+
+This session will show you how to cache globally, i.e. across user sessions
+
+#### Streamlit Global Cache Example
+
+```python
+from time import sleep
+
+import numpy as np
+import streamlit as st
+from matplotlib.figure import Figure
+
+@st.cache_resource
+def get_data():
+    print("get_data func")
+    sleep(1.0)
+    return np.random.normal(1, 1, size=100)
+
+@st.cache_data(hash_funcs={Figure: lambda _: None})
+def plot(data, bins):
+    print("plot func", bins)
+    sleep(2.0)
+    fig = Figure(figsize=(8,4))
+    ax = fig.subplots()
+    ax.hist(data, bins=bins)
+    return fig
+
+data = get_data()
+bins = st.slider(value=20, min_value=10, max_value=30, step=1, label="Bins")
+st.pyplot(plot(data, bins))
+```
+
+The global cache `st.cache_resource` is used on the `get_data` function to only load the data once
+across all users sessions.
+
+#### Panel Global Cache Example
+
+```python
+from time import sleep
+
+import numpy as np
+import panel as pn
+from matplotlib.figure import Figure
+
+
+def get_data():
+    print("data func")
+    sleep(1.0)
+    return np.random.normal(1, 1, size=100)
+
+def plot(data, bins):
+    print("plot func", bins)
+    sleep(2.0)
+    fig = Figure(figsize=(8,4))
+    ax = fig.subplots()
+    ax.hist(data, bins=bins)
+    return fig
+
+pn.extension(sizing_mode="stretch_width", template="bootstrap")
+
+data = pn.state.as_cached(key="data", fn=get_data) # Global Cache
+bins = pn.widgets.IntSlider(value=20, start=10, end=30, step=1)
+cplot = pn.cache(plot) # Session Cache
```

Comment:
`pn.cache` only works for a session. And `pn.state.as_cached` only works for a single value.

How would you recommend I turn the line `cplot=pn.cache(plot)` into something that makes the function cache globally, i.e. across all user sessions @philippjfr ? Are we not missing the utility function to do that?

Reply:
>pn.cache only works for a session

This is news to me. What you're asking should already be the case.

Reply:
I will report it as a bug then

Reply:
I have reported this as a bug in https://github.com/holoviz/panel/issues/5109